src/main/java/org/ambraproject/rhino/service/impl/IngestionService.java

```diff
@@ -78,18 +83,31 @@ private IngestPackage createIngestPackage(Archive archive) throws IOException {
     ArticleXml parsedArticle = new ArticleXml(document);
     ArticleCustomMetadata customMetadata = customMetadataExtractorFactory.parse(document).build();
 
-    // TODO: Allow bucket name to be specified as an ingestion parameter
-    String destinationBucketName = runtimeConfiguration.getCorpusStorage().getDefaultBucket();
 
-    ArticlePackage articlePackage = new ArticlePackageBuilder(destinationBucketName, archive,
-        parsedArticle, manifestXml).build();
+    ArticlePackage articlePackage = new ArticlePackageBuilder(checkBucketName(bucketName),
+        archive, parsedArticle, manifestXml).build();
 
     articlePackage.validateAssetCompleteness(parsedArticle.findAllAssetNodes().getDois());
 
     ArticleMetadata articleMetadata = parsedArticle.build();
     return new IngestPackage(articlePackage, articleMetadata, customMetadata);
   }
 
+  private String checkBucketName(Optional<String> bucketName) {
```

Comment:
That's a good point about the method name. But I think I prefer to keep it as one method so we don't have to do `runtimeConfiguration.getCorpusStorage()` in both places. I added some Javadoc to make sense out of it; do you think 8d90627 is good enough?